### PR TITLE
std: Remove fortune cookies from fatal runtime errors

### LINKED ISSUE #13871: "hello world" contains Lovecraft quotes
Yes, really

### PR DESCRIPTION
**Pull request by brson**
Closes #13871

### CODE REVIEW

**Comment by jfager** [comment]
Boooooo

**Comment by Gankro** [comment]
Instantly r+'d by the PR author... this kind of totalitarian management of the project won't stand!

:smiling_imp:

**Comment by CaptainHayashi** [comment]
I'll always remember coming back to an attempt to compile `rustc`, seeing `You've met with a terrible fate, haven't you?` and being a mixture of annoyed, confused, and (I'll admit) slightly amused xD

**Comment by csouth3** [comment]
:cry: rip.  I only got a Lovecraft quote once ever, but it was pretty much my favorite Rust moment ever.

File: src/libstd/rt/util.rs [hunk 1]
```diff
@@ -144,56 +144,6 @@ pub fn abort(args: fmt::Arguments) -> ! {
     let _ = write!(&mut w, "{}", args);
     let msg = str::from_utf8(&w.buf[0..w.pos]).unwrap_or("aborted");
     let msg = if msg.is_empty() {"aborted"} else {msg};
```
Comment (eddyb) [inline src/libstd/rt/util.rs]: All this logic is not necessary anymore.

Comment (brson) [inline src/libstd/rt/util.rs]: @eddyb It seems to cover real cases: the first if the buffer is not utf8 (though with fmt changing to require utf8 maybe that's impossible), and the second if the abor t message is empty. Neither of those seem to be related to the easter egg.

Comment (eddyb) [inline src/libstd/rt/util.rs]: Sorry, I meant the whole writing to a buffer thing. That was only done for the fortune hash. Can just write the format args to stderr now.

**Comment by tshepang** [comment]
I was surprised that @bors allows changes to be approved by their authors.

**Comment by retep998** [comment]
I'm against this change. If anything we should be encrypting the text so that they are truly Easter eggs.

**Comment by Gankro** [comment]
@tshepang: bors also allows r=litterallyAnything; it is not built to have any particular security model other than "I listen to those with authority".

**Comment by stuartpb** [comment]
:clap: :+1: :clap:

**Comment by alexchandel** [comment]
Why couldn't we just remove the text for non-debug builds?

**Comment by kballard** [comment]
I sympathize with the "2k is significant on some platforms" argument, but I am rather saddened to see this go.

I do wish it could exist only for non-debug builds. Though offhand I'm not sure how that would be possible. I'm also confused as to how this was ending up in `#![no_std]` builds since it apparently was part of `libstd/rt/util.rs`, which one would think is only present in libstd.

**Comment by liigo** [comment]
Nice to see *useless* code gone!

**Comment by monperrus** [comment]
FTR, this PR is mentioned in [Long Live Software Easter Eggs (ACM Queue, 2022)](https://queue.acm.org/detail.cfm?id=3534857)

> "be careful, this battle is epic and dangerous. H.P. Lovecraft was killed by a gunshot in the trenches of the Rust compiler."

**Pull request merged**